### ISSUE TITLE
[BUG] Random errors with docker compose up -d --force-recreate in version 2.19.0

### ISSUE THREAD

**Issue by jurgenhaas** [docker/compose#10748]
### Description

In our build and deploy pipelines we're using docker 24.0.2 and docker compose 2.19.0 where we see random errors since we've updated docker compose from 2.18.1 - before no errors at all, with hundreds of pipelines running every day.

```
$ docker compose up -d --force-recreate --quiet-pull --remove-orphans
 Container drupal-9_main-alerta-1  Recreate
 Container drupal-9_main-redis-1  Recreate
 Container drupal-9_main-mariadb-1  Recreate
 Container drupal-9_main-borgmatic-1  Recreate
 Container drupal-9_main-cron-1  Recreate
 Container drupal-9_main-php-1  Recreate
 Container drupal-9_main-cron-1  Recreated
 Container drupal-9_main-redis-1  Recreated
 Container drupal-9_main-php-1  Recreated
 Container drupal-9_main-apache-1  Recreate
Error response from daemon: container 70979001dd6e8a60ec6ecc7c83eaee8c87f551d1ea6efc92a771d37fd9283fed is not connected to the network traefik-public
```

When going to the host where this happened, we're seeing this status:

```
$ docker compose ps
NAME                                     COMMAND                  SERVICE             STATUS              PORTS
30f91d2adcbd_drupal-9_main-borgmatic-1   "/entry.sh"              borgmatic           created             
591391f7b262_drupal-9_main-mariadb-1     "/docker-entrypoint.…"   mariadb             created             
843a6249ea7e_drupal-9_main-apache-1      "/docker-entrypoint.…"   apache              created             
c63c713edb25_drupal-9_main-alerta-1      "/bin/sh -c 'python …"   alerta              created             
drupal-9_main-alerta-1                   "/bin/sh -c 'python …"   alerta              exited (137)        
drupal-9_main-apache-1                   "/docker-entrypoint.…"   apache              running             80/tcp
drupal-9_main-borgmatic-1                "/entry.sh"              borgmatic           exited (137)        
drupal-9_main-cron-1                     "/usr/bin/ofelia dae…"   cron                created             
drupal-9_main-mariadb-1                  "/docker-entrypoint.…"   mariadb             exited (0)          
drupal-9_main-php-1                      "/docker-entrypoint.…"   php                 created             
drupal-9_main-redis-1                    "/docker-entrypoint.…"   redis               created 
```

So, it looks as if containers are created with a hash-prefix but one of them won't be attached to the configured network. Then that breaks the process. When running `docker compose down` followed by `docker compose up -d` again, the status will be correct and looks like this:

```
NAME                        COMMAND                  SERVICE             STATUS              PORTS
drupal-9_main-alerta-1      "/bin/sh -c 'python …"   alerta              running             
drupal-9_main-apache-1      "/docker-entrypoint.…"   apache              running             80/tcp
drupal-9_main-borgmatic-1   "/entry.sh"              borgmatic           running             
drupal-9_main-cron-1        "/usr/bin/ofelia dae…"   cron                running             
drupal-9_main-mariadb-1     "/docker-entrypoint.…"   mariadb             running             3306/tcp
drupal-9_main-php-1         "/docker-entrypoint.…"   php                 running             9000/tcp
drupal-9_main-redis-1       "/docker-entrypoint.…"   redis               running             6379/tcp
```

We've now downgraded to docker compose 2.18.1 again and that error disappeared.

### Compose Version

```Text
2.19.0
```

### Docker Environment

```Text
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.10.5
    Path:     /usr/local/lib/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.19.0
    Path:     /usr/local/lib/docker/cli-plugins/docker-compose

Server:
 Containers: 211
  Running: 4
  Paused: 0
  Stopped: 207
 Images: 249
 Server Version: 24.0.2
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3dce8eb055cbb6872793272b4f20ed16117344f8
 runc version: v1.1.7-0-g860f061
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
 Kernel Version: 5.4.0-152-generic
 Operating System: Ubuntu 20.04.6 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 31.22GiB
 Name: jhworkstation
 ID: LD2R:J56O:LHV2:ECO4:VNLP:3XKM:M3X7:IINK:M2YA:IQ75:SP5C:MOGB
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Default Address Pools:
   Base: 169.254.0.0/16, Size: 24
   Base: 172.17.0.0/16, Size: 24
   Base: 172.80.0.0/16, Size: 24

WARNING: No swap limit support
```

**Comment by milas** [comment]
Hi, thanks for the report! This is tracked as #10668 (and I'm working on a fix over in #10756).

I'm closing this issue as a duplicate, feel free to follow the others for updates :)

**Comment by jurgenhaas** [comment]
Thanks a lot @milas , much appreciated. Sorry that I opened a duplicate, wasn't able to find the other one in the first place. Looking harder next time ;-)
